### PR TITLE
Splits placeontop proc

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
I find the proc hard to read honestly. There's no reason we can't split this into two functions - the secondary functionality is used only once, in reader.dmm.
## Why It's Good For The Game
Code improvement
Glorious snake case
## Changelog
N/A nothing player facing

### CODE REVIEW

File: code/modules/mapping/reader.dm [hunk 1]
```diff
@@ -950,7 +950,7 @@ GLOBAL_LIST_EMPTY(map_model_default)
 
 		// Note: we make the assertion that the last path WILL be a turf. if it isn't, this will fail.
 		if(placeOnTop)
-			instance = crds.PlaceOnTop(null, members[index], CHANGETURF_DEFER_CHANGE | (no_changeturf ? CHANGETURF_SKIP : NONE))
+			instance = crds.load_on_top(members[index], CHANGETURF_DEFER_CHANGE | (no_changeturf ? CHANGETURF_SKIP : NONE))
```
Comment (jlsnow301) [inline code/modules/mapping/reader.dm:953]: this is the only place that uses the second argument

File: code/game/turfs/baseturfs.dm [hunk 2]
```diff
@@ -42,54 +42,42 @@
...
-			newT.baseturfs = baseturfs_string_list(old_baseturfs + (newT.baseturfs - GLOB.blacklisted_automated_baseturfs), newT)
-			return newT
-		if(!length(baseturfs))
-			baseturfs = list(baseturfs)
-		if(!isclosedturf(src))
-			new_baseturfs = list(type) + new_baseturfs
-		baseturfs = baseturfs_string_list(baseturfs + new_baseturfs, src)
-		return ChangeTurf(fake_turf_type, null, flags)
+
+	var/turf/newT
```
Comment (san7890) [inline code/game/turfs/baseturfs.dm:65]: if you're so insistent on snake casing stuff why not change this one to `new_turf`?

**Review by san7890 — CHANGES_REQUESTED** [review]
potential way to make this pr even better

**Review by Ghommie — APPROVED** [review]
I've read and re-read it a few times, but I'm still hoping that nothing subtly breaks out of this.

**Comment by Y0SH1M4S73R** [comment]
> I've read and re-read it a few times, but I'm still hoping that nothing subtly breaks out of this.

#81099

**Pull request merged**